ibis/odbc/impala.py

```
@@ -0,0 +1,112 @@
+    """
+    Using ODBC  to create ImpalaClient connection.
+
+    Parameters
+    ----------
+    dsn : str, optional
+        Data source name as given in the (unix) odbc.ini file or (Windows)
+        ODBC Data Source Administrator tool.
+    connection_string : str, optional
+        Preformatted ODBC connection string. Specifying this and dsn at the
+        same time raises ParameterError
+    turbodbc_options : dict, optional
+        to be passed to turbodbc_options
+
+    Returns
+    -------
+    ImpalaClient
+    """
+
+    con = ImpalaODBCConnection(pool_size=pool_size,
+                               connection_string=connection_string,
+                               dsn=dsn,
+                               turbodbc_options=turbodbc_options)
+
+    try:
+        client = ImpalaClient(con, hdfs_client=hdfs_client)
+
+        if options.default_backend is None:
+            options.default_backend = client
+    except:
+        con.close()
+        raise
+
+    return client
+
+
+class ImpalaODBCConnection(ImpalaConnection):
+    def _new_cursor(self):
+        params = self.params.copy()
+
+        options = turbodbc.make_options(autocommit=True,
+                                        **params['turbodbc_options'])
+        con = turbodbc.connect(dsn=params['dsn'],
+                               connection_string=params['connection_string'],
+                               turbodbc_options=options)
+
+        self._connections.add(con)
+
+        cursor = con.cursor()
+
+        wrapper = ImpalaODBCCursor(cursor, self, con, self.database,
+                                   self.options.copy())
+        wrapper.set_options()
+
+        return wrapper
+
+    def ping(self):
+        self._get_cursor()
+
+    def execute(self, query, async=False):
+        if isinstance(query):
+            query = query.compile()
+
+        cursor = self._get_cursor()
+        self.log(query)
+
+        try:
+            cursor.execute(query)
+        except:
```

Is there any reason not to just let the exception be raised in the usual manner?